backend/remote-state/swift/backend_test.go

```diff
@@ -0,0 +1,84 @@
+package swift
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/containers"
+	// "github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects"
+	"github.com/hashicorp/terraform/backend"
+)
+
+// verify that we are doing ACC tests or the Swift tests specifically
+func testACC(t *testing.T) {
+	skip := os.Getenv("TF_ACC") == "" && os.Getenv("TF_SWIFT_TEST") == ""
+	if skip {
+		t.Log("swift backend tests require setting TF_ACC or TF_SWIFT_TEST")
+		t.Skip()
+	}
+	t.Log("swift backend acceptance tests enabled")
+}
+
+func TestBackend_impl(t *testing.T) {
+	var _ backend.Backend = new(Backend)
+}
+
+func testAccPreCheck(t *testing.T) {
+	v := os.Getenv("OS_AUTH_URL")
+	if v == "" {
+		t.Fatal("OS_AUTH_URL must be set for acceptance tests")
+	}
+}
+
+func TestBackendConfig(t *testing.T) {
+	testACC(t)
+
+	// Build config
+	config := map[string]interface{}{
+		"path": "test-tfstate",
+	}
+
+	b := backend.TestBackendConfig(t, New(), config).(*Backend)
+
+	if b.path != "test-tfstate" {
+		t.Fatalf("Incorrect path was provided.")
+	}
+}
+
+func TestBackend(t *testing.T) {
+	testACC(t)
+
+	path := fmt.Sprintf("terraform-state-swift-test-%x", time.Now().Unix())
+
+	b := backend.TestBackendConfig(t, New(), map[string]interface{}{
+		"path": path,
+	}).(*Backend)
+
+	// createSwiftContainer(t, b.client, path)
```

Comment:
No worries... I'm guna take a stab at the additional tests for `archive_path` and `expire_after` aswell...